### PR TITLE
[openstack_cinder] check for api service running via cinder_wsgi

### PR DESCRIPTION
**Pull request by stuggi**
With OSP11 cinder api changed to run via https wsgi. To check for
running cinder-manage command we also need to take this situation.
The change checks for cinder_wsgi process.

Signed-off-by: Martin Schuppert <mschuppert@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

File: sos/plugins/openstack_cinder.py [hunk 1]
```diff
@@ -37,14 +37,22 @@ def setup(self):
...
+                if "cinder_wsgi" in line:
+                    in_ps = True
+                    break
+
         container_status = self.get_command_output("docker ps")
         in_container = False
         if container_status['status'] == 0:
             for line in container_status['output'].splitlines():
                 if line.endswith("cinder_api"):
                     in_container = True
```
Comment (pmoravec) [inline sos/plugins/openstack_cinder.py]: Patch sounds OK, maybe trivial enhancement outside of its scope would be calling "break" here (like on line 46)?

Comment (stuggi) [reply sos/plugins/openstack_cinder.py]: yes makes sense. I submitted a change with the additional "break"

File: sos/plugins/openstack_cinder.py [hunk 2]
```diff
@@ -37,14 +37,23 @@ def setup(self):
...
+                    break
 
-        if (service_status['status'] == 0) or in_container:
+        if (service_status['status'] == 0) or in_container or in_ps:
```
Comment (bmr-cymru) [inline sos/plugins/openstack_cinder.py]: Do we really need three separate conditions to enable this? And why is there a test for `in_container` inside the block? This does not seem all that readable or optimal right now.

Why not place the set up of `cinder_config` outside this block, simplifying it and placing the `cinder-manage` command collection closer to the condition that enables it?

I suspect that would also allow the condition to be simplified.

File: sos/plugins/openstack_cinder.py [hunk 3]
```diff
@@ -37,14 +37,23 @@ def setup(self):
             "systemctl status openstack-cinder-api.service"
         )
 
+        ps_status = self.get_command_output("ps -ef")
```
Comment (bmr-cymru) [inline sos/plugins/openstack_cinder.py]: I'm not really keen on this in plugins. The model we talked about some time ago (but which still is not implemented) is for a set of "core" plugins to collect this type of information and other plugins can then use it in their work. That isn't ready today, and is a very considerable piece of work, but I would prefer it if we thought about how we want things to look now, and carried out changes like this in a way that aligns to that future direction.

What I'd like  to see is a new `Plugin` method that calls `ps` and perform the search. Then if/when we ever move to the rich plugin model that method simply becomes a way to access the ps data cached by the process plugin (or whatever).

Cramming it into individual plugins invites proliferation and makes it more likely the next plugin author to need something like this will invent their own wheel again...

File: sos/plugins/openstack_cinder.py (hunk 2 above)
Comment (stuggi) [reply sos/plugins/openstack_cinder.py]: I reviewed that part and removed one condition by checking the ps for either cinder-api or cinder_wsgi. with this we can remove one of the conditions. 

Also moved the cinder_config part in when we check for running cinder in a container env.

File: sos/plugins/openstack_cinder.py (hunk 3 above)
Comment (stuggi) [reply sos/plugins/openstack_cinder.py]: I have introduced a new method check_process_by_name in my last update within this PR. Let me know if I should do it in a separate PR as it is a general enhancement or if it is ok within this PR.

File: sos/plugins/__init__.py [hunk 4]
```diff
@@ -1026,6 +1026,17 @@ def report(self):
         else:
             return html
 
+    def check_process_by_name(self, pr):
```
Comment (bmr-cymru) [inline sos/plugins/__init__.py]: This adds an entire new interface to the `Plugin` class: it needs to be in a separate commit with an adequate explanation in the commit message.

File: sos/plugins/__init__.py [hunk 5]
```diff
@@ -1026,6 +1026,17 @@ def report(self):
...
+        """Checks if a named process is listed in ps -ef output."""
```
Comment (bmr-cymru) [inline sos/plugins/__init__.py]: That's not actually what this method does: it returns `True` if the given `pr` string is a *substring* of any running process's `ps` "full format" listing.. i.e. command name, arguments, etc. That's probably **not** what is wanted for this interface.

I'm dubious that the `ps` command is even the best choice of tool here. We could walk the process tree ourselves (not that costly when just checking names), or use a more focused tool like `pgrep`.

File: sos/plugins/__init__.py (hunk 4 above)
Comment (bmr-cymru) [reply sos/plugins/__init__.py]: The interface itself is fine, although I would suggest using something like `process`, `name`, or `command` instead of `pr` - there's only a single method argument here after self, so we're not getting short on space and API argument names should be as descriptive as possible where space allows.

Comment (stuggi) [reply sos/plugins/__init__.py]: submitted https://github.com/sosreport/sos/pull/1186 for the new method
